### PR TITLE
gtk-layer-shell: fix cross compilation

### PR DESCRIPTION
**Pull request by NickCao**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] riscv64-linux (cross)
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 211502` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>21 packages built:</summary>
  <ul>
    <li>avizo</li>
    <li>eww-wayland</li>
    <li>greetd.gtkgreet</li>
    <li>gtk-layer-shell</li>
    <li>gtklock</li>
    <li>nwg-bar</li>
    <li>nwg-dock</li>
    <li>nwg-drawer</li>
    <li>nwg-launchers</li>
    <li>nwg-menu</li>
    <li>nwg-panel</li>
    <li>nwg-wrapper</li>
    <li>sirula</li>
    <li>swaynotificationcenter</li>
    <li>swaysettings</li>
    <li>waybar</li>
    <li>wayfire</li>
    <li>wayfirePlugins.wf-shell (wayfireApplications-unwrapped.wayfirePlugins.wf-shell)</li>
    <li>wayfireApplications-unwrapped.wcm</li>
    <li>wcm</li>
    <li>wlogout</li>
  </ul>
</details>

File: pkgs/development/libraries/gtk-layer-shell/default.nix [hunk 1]
```diff
@@ -35,11 +52,12 @@ stdenv.mkDerivation rec {
     docbook-xsl-nons
     docbook_xml_dtd_43
     vala
+    wayland-scanner
   ];
 
   buildInputs = [
-    wayland
     gtk3
+    gtk-doc
```
Comment (Artturin) [inline pkgs/development/libraries/gtk-layer-shell/default.nix:60]: https://github.com/wmww/gtk-layer-shell/blob/3073ff944ad19e8c3db5243bb9079f3fc54a23cb/doc/meson.build#L7 should be removed. `gnome.gtkdoc` already does its own checking

**Comment by Artturin** [comment]
Darwin build fails

```
meson.build:15:0: ERROR: Dependency "wayland-client" not found, tried pkgconfig and framework
```

**Comment by NickCao** [comment]
Still fails, is wayland even supported on darwin?

**Comment by Artturin** [comment]
checked and `wayland` does not contain `wayland-client.pc` on darwin, only `wayland-scanner.pc`
also checked that `gtk-layer-shell` was failing before this pr

lets mark it linux only

**Review by superherointj — APPROVED** [review]

**Pull request merged**